lld/ELF/Relocations.cpp

```diff
@@ -910,14 +911,26 @@ void elf::addGotEntry(Ctx &ctx, Symbol &sym) {
 
```

Comment:
This is too much pauth specific code in the generic addGotEntry. What if we change the callers to call a separate function for pauth?

Reply:
Moved pauth-specific logic to `addGotAuthEntry`, thanks for suggestion, see e1f82608113396911761f0b97dd57ccd59a16448.

I've updated the caller-side code in `elf::postScanRelocations` to check against `NEEDS_GOT_AUTH`. There is also a call to `addGotEntry` in `handleNonPreemptibleIfunc`, but, we anyway do not support `R_AARCH64_AUTH_IRELATIVE`, so just added a corresponding assertion checking absence of `NEEDS_GOT_AUTH` (see d7336f62cb65ac40d0f157f24838ada9ae32ebe3). Probably the assertion message could be re-phrased somehow - suggestions are welcome :)